scripts/runcode.jl

```diff
@@ -329,28 +350,59 @@ function chamber(
     log_vfr_vector::Union{Float64,Vector{Float64}},
     depth_vector::Union{Float64,Vector{Float64}},
     output_dirname::String=get_timestamp(),
+    ;
+    plotfig::Bool=true,
 )::String
+    check_for_duplicates(
+        log_volume_km3_vector,
+        InitialConc_H2O_vector,
+        InitialConc_CO2_vector,
+        log_vfr_vector,
+        depth_vector,
+    )
     path0 = joinpath(pwd(), output_dirname)
     mkdir(path0)
     to = TimerOutput()
+    df_outputs = Vector{ChamberOutput}(
+        undef,
+        length(log_volume_km3_vector) *
+        length(InitialConc_H2O_vector) *
+        length(InitialConc_CO2_vector) *
+        length(log_vfr_vector) *
+        length(depth_vector),
+    )
     @timeit to "chamber" begin
-        for log_volume_km3 in log_volume_km3_vector,
-            InitialConc_H2O in InitialConc_H2O_vector,
-            InitialConc_CO2 in InitialConc_CO2_vector,
-            log_vfr in log_vfr_vector,
-            depth in depth_vector
-
+        Threads.@threads for (
+            idx, (log_volume_km3, InitialConc_H2O, InitialConc_CO2, log_vfr, depth)
+        ) in collect(
+            enumerate(
+                Iterators.product(
+                    log_volume_km3_vector,
+                    InitialConc_H2O_vector,
+                    InitialConc_CO2_vector,
+                    log_vfr_vector,
+                    depth_vector,
+                ),
+            ),
+        )
             dataset = "vol$(log_volume_km3)_h2o$(InitialConc_H2O)_gas$(InitialConc_CO2)_vfr$(log_vfr)_dep$(depth)"
-            chamber(
+            df = chamber(
                 composition,
                 end_time,
                 log_volume_km3,
                 InitialConc_H2O,
                 InitialConc_CO2,
                 log_vfr,
                 depth,
-                joinpath(output_dirname, dataset),
+                joinpath(output_dirname, dataset);
+                plotfig=false,
             )
+            df_outputs[idx] = ChamberOutput(df, joinpath(output_dirname, dataset))
+        end
+        if plotfig
+            for output in df_outputs
+                plot_figs(output.df, output.path)
+            end
         end
```

Comment:
`df_outputs` stored the `dataframe` and `path` of each dataset during the computation. Then generate the plots after the computation finished.